endpoint/oncall.go

```diff
@@ -0,0 +1,173 @@
+package endpoint
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/opsctl/service/opsgenie"
+
+	"github.com/giantswarm/auto-oncall/service/githubhook"
+)
+
+const (
+	masterRef       = "refs/heads/master"
+	listenAddres    = 8000
+	webhookEndpoint = "/webhook"
+	routingRuleTTL  = time.Hour * time.Duration(1)
+	routingRuleType = "match-all-conditions"
+)
+
+type healthCheckResponse struct {
+	Status string `json:"status"`
+}
+
+type errorResponse struct {
+	Error string `json:"error"`
+}
+
+type Config struct {
+	Logger        micrologger.Logger
+	OpsgenieToken string
+	Repositories  []string          `yaml:"repositories"`
+	Users         map[string]string `yaml:"users"`
+	WebhookSecret string
+}
+
+type Oncall struct {
+	logger        micrologger.Logger
+	opsgenieToken string
+	repositories  []string
+	users         map[string]string
+	webhookSecret []byte
+}
+
+func New(c Config) (Oncall, error) {
+	if c.OpsgenieToken == "" {
+		return Oncall{}, microerror.Maskf(invalidConfigError, "OPSGENIE_TOKEN environment variable token must not be empty")
+	}
+	if c.WebhookSecret == "" {
+		return Oncall{}, microerror.Maskf(invalidConfigError, "GITHUB_WEBHOOK_SECRET environment variable must not be empty")
+	}
+
+	oncall := Oncall{
+		logger:        c.Logger,
+		opsgenieToken: c.OpsgenieToken,
+		repositories:  c.Repositories,
+		users:         c.Users,
+		webhookSecret: []byte(c.WebhookSecret),
+	}
+
+	return oncall, nil
+}
```

Comment:
Ordering again:

- Type (Oncall)
- Constructors (New)
- Public methods (NewServer)
- Private methods (webhookHandler, createRoutingRule)
- Helper functions (the rest)